test/Microsoft.SqlTools.ServiceLayer.UnitTests/ObjectExplorer/NodeFilterTests.cs

```diff
@@ -610,5 +613,307 @@ public void TextNumericFilters()
             };
             Assert.Throws<IndexOutOfRangeException>(() => INodeFilter.GetPropertyFilter(filterList, typeof(SqlHistoryTableQuerier), ValidForFlag.All), "Error not thrown when the array contains single value for between operator");
         }
+
+        [Test]
+        public void TestContainsFilter()
+        {
+            // Testing text filter with ends with operator
+            var filterList = new List<NodePropertyFilter>
```

Comment:
I don't think I've ever seen a response to this - do you have any plans to add tests that actually run the filter through STS to verify it's being built correctly? Right now this doesn't actually tell us whether the queries generated are valid or not.

Reply:
You mean Integration tests?

Reply:
That'd probably be required - I don't believe SMO has a way to filter without a live connection.

@shueybubbles Any thoughts here on what the best way to get this kind of test added would be? (in case it's not clear, these filter strings are added to the SMO URN used to fetch the objects)

Reply:
We have ways to get a live connection - I just wanted to ask if you knew of a way to do an offline test of the filtering functionality. IIRC though these filter queries are turned into actual T-SQL queries so that's probably not possible. Is that correct?

Reply:
yeah there's a bunch of code in SFC that converts the URN snippets to TSQL snippets. I think that area is only tested indirectly not exhaustively in the SMO suites. You might consider putting your tests for correct SQL query generation in SMO itself to prevent regressions there.